### DISCUSSION TITLE
Winforms to AlternetUI migration

### DISCUSSION THREAD

**Discussion by neoxeo** [alternetsoft/AlternetUI#126] [Ideas]
@generalloki 

I continue to work slowly on my tool to help users to migrate Winforms apps to AlternetUI apps.

Here is a small Winforms sample project (framework 4.8) I use to test migration if it can be useful for you : 
[WindowsForms_Test1_15.12.2023.zip](https://github.com/alternetsoft/AlternetUI/files/13681602/WindowsForms_Test1_15.12.2023.zip)
I will convert it to .NET8.0 to have two projects for tests

**Today, these components have not same name as Winforms or not exist :** 
```
FlowLayoutPanel flowLayoutPanel = new FlowLayoutPanel();
MaskedTextBox maskedTextBox = new MaskedTextBox();
MenuStrip menuStrip = new MenuStrip();
PrintPreviewControl printPreviewControl = new PrintPreviewControl();
SplitContainer splitContainer = new SplitContainer();
Splitter splitter = new Splitter();
TableLayoutPanel tableLayoutPanel = new TableLayoutPanel();
ToolStripContainer toolStripContainer = new ToolStripContainer();
```

If components exist in AlternetUI but not have the same name, can you give me the equivalent ?

- For ToolTip, it exist but have error when I use : 
ToolTip toolTip = new ToolTip();

- For VScrollBar and HScollBar I have seen it is now present since this morning :-) , thanks !

I will post progress here.

**Comment by neoxeo** [comment]
- `MonthCalendar` in Winforms is equivalent to `Calendar` in AlternetUI
- `FolderBrowserDialog` in Winforms is equivalent to `SelectDirectoryDialog` in AlternetUI
- `TrackBar` in Winforms is equivalent to `Slider` in AlternetUI
- `DataGridView` in Winforms is equivalent to `Grid` in AlternetUI
- `StatusStrip` in Winforms is equivalent to `StatusBar` in AlternetUI
- `ToolStrip` in Winforms is equivalent to `Toolbar` in AlternetUI

**Comment by generalloki** [comment]
There are no MaskedTextBox, Splitter controls in UI.
DataGridView is not an equivalent to Grid in UI, it is different control and there is no alternative in UI.

**Comment by generalloki** [comment]
I am moving this to discussions. I beleive we need to have separate issue for the property/method as not all of them are easy to implement and there are currently more important things to do in the project.
